File: billing-domain/src/main/java/com/example/billing/domain/payment/PaymentStatus.java
package com.example.billing.domain.payment;

public enum PaymentStatus {
    PENDING,    // PG 호출 전
    APPROVED,   // PG 응답 OK
    FAILED;     // PG 응답 실패 (Circuit open / 사용자 취소 등)

    public boolean isTerminal() { return this == APPROVED || this == FAILED; }
}

File: billing-domain/src/main/kotlin/com/example/billing/domain/payment/PaymentStatus.kt
package com.example.billing.domain.payment

enum class PaymentStatus {
    /** PG 호출 전 */
    PENDING,

    /** PG 응답 OK */
    APPROVED,

    /** PG 응답 실패 (Circuit open / 사용자 취소 등) */
    FAILED;

    fun isTerminal(): Boolean = this == APPROVED || this == FAILED
}
